WalletWasabi.Tests/UnitTests/Crypto/ProofSystemTests.cs

```diff
@@ -69,15 +65,65 @@ public void CanProveAndVerifySerialNumbers()
 			var r = rnd.GetScalar();
 			var a = rnd.GetScalar();
 
-			var witness = new ScalarVector(z, a, r);
-			var Ca = z * Generators.Ga + r * Generators.Gh + a * Generators.Gg;
-			var S = r * Generators.Gs;
+			static (GroupElement Ca, GroupElement S) SerialNumberPublicPoints(Scalar z, Scalar a, Scalar r) =>
```

Comment:
this should go in `ProofSystem.cs`, we will need it in the real code. if it gets too busy we can move parts to a separate WabiSabi class